### PR TITLE
Multi board

### PR DESCRIPTION
**Pull request by neokry**
Updates contracts and subgraph to support cloning
1. Renames main contract to Kanban for simplicity
2. Creates KanbanFactory to generate Kanban contracts
3. Updates subgraph to get data from both contracts.

### CODE REVIEW

File: packages/hardhat/contracts/KanbanFactory.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  address private baseKanbanAddress;
+  
+  constructor(address _baseKanbanAddress) {
+    baseKanbanAddress = _baseKanbanAddress;
+  }
+
+  function createKanban() public {
+    address instance = Clones.clone(baseKanbanAddress);
+    kanbanInstances[_kanbanIds.current()] = instance;
+    emit kanbanCreated(_kanbanIds.current(), msg.sender, instance);
```
Comment (sazerac-eth) [inline packages/hardhat/contracts/KanbanFactory.sol:26]: (Sorry if this is totally wrong) 
I think this introduces a re-entrancy vulnerability. E.g. if the contract is called multiple times quickly we'll create multiple kanbans and lose some of them. 

In general programming I would have said that having a predictable increasing ID is a bad thing too, but I guess that whatever ID we pick, it's all going to be visible on the chain anyway :) 

But maybe using a securely generated random ID here would avoid collisions.

File: packages/hardhat/test/kanban-factory-test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const chai = require("chai");
```
Comment (sazerac-eth) [inline packages/hardhat/test/kanban-factory-test.js]: Looks like you aren't using this line

File: packages/hardhat/test/kanban-factory-test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    await this.kanban.deployed();
+
+    this.kanbanFactory = await this.KanbanFactory.deploy(this.kanban.address);
+    await this.kanbanFactory.deployed();
+  });
+
+  it("Should create new kanbans", async function () {
+    await this.kanbanFactory.createKanban();
+    const instance = this.kanbanFactory.kanbanInstances(0);
+    expect(instance).is.not.null;
```
Comment (sazerac-eth) [inline packages/hardhat/test/kanban-factory-test.js:22]: This isn't null, but it is empty (`{}`).

**Review by sazerac-eth — APPROVED** [review]
Looks generally good, with two things that might need fixing: 

1. Your test which asserts that `instance` exists should really be failing, because instance is empty.  
2. There's a potential re-entrancy bug. 

I also can't really review the subgraph stuff, it's my first time using it. But logically I can't spot any problems. I noted some potential issues but if this is all working then it gets my approval 👍  

Good work ⚔️

File: packages/hardhat/contracts/KanbanFactory.sol (hunk 1 above)
Comment (rpal3058) [reply packages/hardhat/contracts/KanbanFactory.sol:26]: "But maybe using a securely generated random ID here would avoid collisions." - 
I think we can simply return or emit the address where its deployed, if the id creates a collision

**Pull request merged**